pkg/db/db.go

```diff
@@ -619,17 +619,19 @@ type DBEnvironmentLock struct {
 	Metadata   string
 }
 
-type EnvironmentLockMetadata struct {
+type LockMetadata struct {
 	CreatedByName  string
 	CreatedByEmail string
 	Message        string
 }
 
 type AllDeployments []Deployment
 type AllEnvLocks map[string][]EnvironmentLock
+type AllAppLocks map[string]map[string][]ApplicationLock // EnvName-> AppName -> []Locks
 
 type GetAllDeploymentsFun = func(ctx context.Context, transaction *sql.Tx) (AllDeployments, error)
 type GetAllEnvLocksFun = func(ctx context.Context, transaction *sql.Tx) (AllEnvLocks, error)
+type GetAllAppLocksFun = func(ctx context.Context, transaction *sql.Tx) (AllAppLocks, error)
```

Comment:
the functions here should not require a transaction. They should read only from the manifest repo.
```suggestion
type GetAllAppLocksFun = func(ctx context.Context) (AllAppLocks, error)
```